src/Workspaces/Core/Portable/Workspace/Workspace_Editor.cs

```diff
@@ -335,7 +355,14 @@ protected void OnDocumentContextUpdated(DocumentId documentId)
 
             if (container != null)
             {
-                OnDocumentContextUpdated(documentId, container);
+                if (_isProjectUnloading)
```

Comment:
We do not have threading asserts, but they both happen on the UI thread in practice...